pkg/system/phase4_configuring.go

```diff
@@ -537,16 +541,156 @@ func (r *Reconciler) awaitEndpointDeploymentPods() error {
 func (r *Reconciler) ReconcileHPAEndpoint() error {
 	// Wait for the the endpoint deployment to become ready
 	// only if HPA was not created yet
+	log := r.Logger.WithField("func", "ReconcileHPAEndpoint")
 	if r.HPAEndpoint.UID == "" {
 		if err := r.awaitEndpointDeploymentPods(); err != nil {
 			return err
 		}
 	}
+	var err error
+	prometheusList := deployedPrometheus(log)
+	switch r.NooBaa.Spec.AutoscalingType {
+	case "keda":
```

Comment:
a general comment - instead of referring to the types as string, use the consts 
```suggestion
	case nbv1.AutoscalingTypeKeda:
```